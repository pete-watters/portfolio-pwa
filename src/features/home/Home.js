import React from 'react';
import Page from 'components/Page';
import SocialMedia from 'components/SocialMedia';
import Link from 'components/Link';

const Home = () =>
  <Page>
    <p>Full stack developer, UX specialist and JavaScript enthusiast.</p>
    <p>Reach me at: <Link href="mailto:pete@cteic.ie" text="pete@cteic.ie" /></p>
    <footer>
      <SocialMedia />
    </footer>
  </Page>;

export default Home;
