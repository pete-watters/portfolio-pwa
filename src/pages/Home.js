import React, { memo } from 'react';
import logo from '../profilepic.jpg';
import Page from '../components/Page';
import SocialMedia from '../components/SocialMedia';
import Link from '../components/Link';

const Home = () =>
  <Page>
      <img src={logo} className="profile" alt="" />
      <p>Full stack developer, UX specialist and JavaScript enthusiast.</p>
      <p>Academically qualified engineer and hobbyist coder turned craftsman.</p>
      <p>I specialise in well tested, clean scalable code with an elegant and intuitive UX.</p>
      <p>Available for hire as a software developer, coach and speaker.</p>
      <p>Reach me at: <Link href="mailto:pete@cteic.ie" text="pete@cteic.ie" /></p>
      <footer>
      <SocialMedia />
      </footer>
  </Page>;
  
export default memo(Home);