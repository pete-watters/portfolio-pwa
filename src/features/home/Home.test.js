/* eslint-disable no-magic-numbers */
import React from 'react';
import { shallow } from 'enzyme';
import Home from './Home';

const component = shallow(<Home />);

describe('Home', () => {
  it('should render the component correctly', () => {
    expect(component).toMatchSnapshot();
  });
  it('renders the correct text', () => {
    expect(component.find('p').at(0).text()).toEqual('Full stack developer, UX specialist and JavaScript enthusiast.');
    expect(component.find('p').at(1).text()).toEqual('Academically qualified engineer and hobbyist coder turned craftsman.');
    expect(component.find('p').at(2).text()).toEqual('I specialise in well tested, clean scalable code with an elegant and intuitive UX.');
  });
});
