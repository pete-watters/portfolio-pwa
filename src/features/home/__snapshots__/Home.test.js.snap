// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render the component correctly 1`] = `
<Memo(Page)>
  <img
    alt=""
    className="profile"
    src="profilepic.jpg"
  />
  <p>
    Full stack developer, UX specialist and JavaScript enthusiast.
  </p>
  <p>
    Academically qualified engineer and hobbyist coder turned craftsman.
  </p>
  <p>
    I specialise in well tested, clean scalable code with an elegant and intuitive UX.
  </p>
  <p>
    Available for hire as a software developer, coach and speaker.
  </p>
  <p>
    Reach me at: 
    <Memo(Link)
      href="mailto:pete@cteic.ie"
      text="pete@cteic.ie"
    />
  </p>
  <footer>
    <SocialMedia />
  </footer>
</Memo(Page)>
`;
